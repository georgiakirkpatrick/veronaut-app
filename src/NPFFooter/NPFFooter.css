.NPFFooter {
    height: 10vh;
    width: 100%;
    padding: 0 5vw;
    background-color: #2F3B51;
    color: #fff;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    text-align: match-parent;
    position: absolute;
    bottom: 0;
    left: 0;
    font: 400 16px Poppins;
    border: none;
    border-radius: 0;
}

.NPFFooter button {
    height: 100%;
    background: rgba(255, 255, 255, 0);
    font: 500 16px/20px Poppins, sans-serif;
    color: #fff;
    border: none;
    margin: 0;
}

.NPFFooter__next {
    text-align: right;
}

.NPFFooter__previous {
    text-align: left;
}

.NPFFooter__next,
.NPFFooter__previous {
    padding: 0;
}

.NPFFooter__left-arrow,
.NPFFooter__right-arrow {
    font: 500 16px/20px;
    letter-spacing: 1px;
    font-weight: 100;
}

.NPFFooter__left-arrow {
    margin-right: 3vw;
}

.NPFFooter__right-arrow {
    margin-left: 3vw;
}

@media screen and (min-width: 768px) {
    .NPFFooter {
        padding: 0 7vw;
    }
}

@media screen and (min-width: 1024px) {
    .NPFFooter {
        padding: 0 7vw;
    }
    
    .NPFFooter button {
        margin: 0;
        width: 30%;
        height: 100%;
    }
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .NPFFooter {
        height: 7vh;
        padding: 0 10vw;
    }
}

@media screen and (min-width: 1440px) {
    .NPFFooter {
        padding: 0 15vw;
    }
}

@media screen and (min-width: 2560px) {
    .NPFFooter {
        padding: 0 25vw;
        height: 10vh;
    }
}