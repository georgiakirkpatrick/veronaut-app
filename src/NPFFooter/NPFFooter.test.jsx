import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import NPFFooter from './NPFFooter'

describe('NPFFooter', () => {
  it ('renders without crashing', () => {
    render(<NPFFooter />)
    screen.debug()
  })

  it ('render the UI as expected', () => {
    const tree = render(<NPFFooter />)
    expect(tree).toMatchSnapshot()
  })
})