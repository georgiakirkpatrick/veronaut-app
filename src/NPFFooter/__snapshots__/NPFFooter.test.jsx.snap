// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NPFFooter > render the UI as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="NPFFooter"
      >
        <button
          class="NPFFooter__previous"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
            />
          </svg>
          <span>
            PREVIOUS
          </span>
        </button>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="NPFFooter"
    >
      <button
        class="NPFFooter__previous"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
          data-icon="angle-left"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            fill="currentColor"
          />
        </svg>
        <span>
          PREVIOUS
        </span>
      </button>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
